src/mrsimulator/spin_system/isotope.py

```diff
@@ -21,6 +24,50 @@
 }
 
 
+class IsotopeReferences:
+    _instance = None
+
+    def __new__(cls, *args, **kwargs):
+        if not cls._instance:
+            cls._instance = super().__new__(cls, *args, **kwargs)
+        return cls._instance
+
+    def __init__(self):
+        if not hasattr(self, "initialized"):  # Ensure __init__ is only called once
+            self.initialized = True
+            self.isotopes_data = REFERENCE_DATA
+
+    def get_isotope_data(self, symbol: str):
+        return self.isotopes_data.get(symbol, None)
+
+
+class IsotopeReference:
+    def __init__(self, symbol, data):
+        self.ratio = data.get("ratio") / 100
+        self.compound = data.get("compound")
+        self.solvent = data.get("solvent")
+        self.concentration = data.get("concentration")
+
+    def __repr__(self):
+        return json.dumps(
+            {
+                "ratio": self.ratio,
+                "compound": self.compound,
+                "solvent": self.solvent,
+                "concentration": self.concentration,
+            },
+            indent=4,
+        )
+
+
+def isotope_reference(symbol):
+    isotopes_instance = IsotopeReferences()
+    data = isotopes_instance.get_isotope_data(symbol)
+    if data:
+        return IsotopeReference(symbol, data)
+    return None
```

Comment:
This function seems to belong as a property of the Isotope class. Consider addding the following to the isotope class.

```py
    @property
    def reference(self):
        """Reference data"""
        data = REFERENCE_DATA.get(self.symbol)
        return IsotopeReference(**data)
```

Reply:
Are you suggesting that we merge the REFERENCE_DATA and ISOTOPE_DATA into the same json file?  Other than H-2, the REFERENCE_DATA is identical for every isotope of a given element.   It seems more efficient to keep the references separate.   But maybe you're suggesting something simpler?

Reply:
I am not suggesting merging json files. I am suggesting instead of a module function `def isotope_reference(symbol)`, it might be better as a `Isotope` class function `def reference(self)`. It is similar to how you added the `def efg_to_Cq(self)` function to Isotope class.

You raised a good point of what if the isotope doesn't have the corresponding reference data. In that case, you can add an exception in the function with message that reference data is not available. Something like:

```py
    @property
    def reference(self):
        """Reference data"""
        data = REFERENCE_DATA.get(self.symbol, None)
        if data is not None:
            return IsotopeReference(**data)
        raise Exception(f"Reference data is not available for {self.isotope} isotope.")
````